client/src/featureform/serving.py

```diff
@@ -387,10 +387,41 @@ def get_input_df(self, source_name, source_variant):
             else:
                 raise ValueError(f"Unsupported file format for {file_path}")
             return df
+        elif (
+            self.db.is_transformation(source_name, source_variant)
+            == SourceType.DIRECTORY
+        ):
+            source = self.db.get_source_variant(source_name, source_variant)
+            directory = source["definition"]
+            return self.read_directory(directory)
+
         else:
             df = self.process_transformation(source_name, source_variant)
         return df
 
+    def read_directory(self, directory):
+        if not os.path.isdir(directory):
+            raise Exception(f"Path {directory} is not a directory")
+
+        def absolute_file_paths(directory):
+            for dirpath, _, filenames in os.walk(directory):
+                for f in filenames:
+                    yield os.path.abspath(os.path.join(dirpath, f)), f
+
+        file_names = []
+        file_body = []
+        for absolute_fn, relative_fn in absolute_file_paths(directory):
+            file_names.append(relative_fn)
+            f = open(absolute_fn, "r")
+            try:
+                file_body.append(f.read())
+            except Exception as e:
+                raise Exception(
+                    f"Cannot read file {absolute_fn}: {e}\nFiles must be text files"
```

Comment:
This error won't always throw due to it being a text file, could also throw if there aren't permissions.